azure/services/securitygroups/securitygroups_test.go

```diff
@@ -322,16 +225,101 @@ func TestDeleteSecurityGroups(t *testing.T) {
 			defer mockCtrl.Finish()
 
 			scopeMock := mock_securitygroups.NewMockNSGScope(mockCtrl)
-			clientMock := mock_securitygroups.NewMockclient(mockCtrl)
+			reconcilerMock := mock_async.NewMockReconciler(mockCtrl)
 
-			tc.expect(scopeMock.EXPECT(), clientMock.EXPECT())
+			tc.expect(scopeMock.EXPECT(), reconcilerMock.EXPECT())
 
 			s := &Service{
-				Scope:  scopeMock,
-				client: clientMock,
+				Scope:      scopeMock,
+				Reconciler: reconcilerMock,
+			}
+
+			err := s.Delete(context.TODO())
+			if tc.expectedError != "" {
+				g.Expect(err).To(HaveOccurred())
+				g.Expect(err).To(MatchError(tc.expectedError))
+			} else {
+				g.Expect(err).NotTo(HaveOccurred())
 			}
+		})
+	}
+}
+
+var (
+	ruleA = network.SecurityRule{
+		Name: to.StringPtr("A"),
+		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+			Description:              to.StringPtr("this is rule A"),
+			Protocol:                 network.SecurityRuleProtocolTCP,
+			DestinationPortRange:     to.StringPtr("*"),
+			SourcePortRange:          to.StringPtr("*"),
+			DestinationAddressPrefix: to.StringPtr("*"),
+			SourceAddressPrefix:      to.StringPtr("*"),
+			Priority:                 to.Int32Ptr(100),
+			Direction:                network.SecurityRuleDirectionInbound,
+		},
+	}
+	ruleB = network.SecurityRule{
+		Name: to.StringPtr("B"),
+		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+			Description:              to.StringPtr("this is rule B"),
+			Protocol:                 network.SecurityRuleProtocolTCP,
+			DestinationPortRange:     to.StringPtr("*"),
+			SourcePortRange:          to.StringPtr("*"),
+			DestinationAddressPrefix: to.StringPtr("*"),
+			SourceAddressPrefix:      to.StringPtr("*"),
+			Priority:                 to.Int32Ptr(100),
+			Direction:                network.SecurityRuleDirectionOutbound,
+		},
+	}
+	ruleBModified = network.SecurityRule{
+		Name: to.StringPtr("B"),
+		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+			Description:              to.StringPtr("this is rule B"),
+			Protocol:                 network.SecurityRuleProtocolTCP,
+			DestinationPortRange:     to.StringPtr("80"),
+			SourcePortRange:          to.StringPtr("*"),
+			DestinationAddressPrefix: to.StringPtr("*"),
+			SourceAddressPrefix:      to.StringPtr("*"),
+			Priority:                 to.Int32Ptr(100),
+			Direction:                network.SecurityRuleDirectionOutbound,
+		},
+	}
+)
 
-			g.Expect(s.Delete(context.TODO())).To(Succeed())
+func TestRuleExists(t *testing.T) {
```

Comment:
What do you think about moving the tests for this helper function to `spec_test.go` since it's defined in `spec.go`?